### PR TITLE
Add DNS provider for Ionos.

### LINKED ISSUE #727: Add support for ionos / 1and1
### Ionos / 1and1.fr Acme DNS challenge support

Do you think it's possible to support ionos (ex 1and1.fr) for Acme DNS-01 challenge ?
I'd love to have a wildcard tls certificate :smile: 

Related issue : https://github.com/containous/traefik/issues/4267

### PR DESCRIPTION
**Pull request by ldez**
- [x] have a homogeneous design with the other providers
- [x] pass the linter
- [x] do `go mod tidy`
- [x] add tests (units)
- [x] add tests ("live")
- [x] add a [provider descriptor](https://github.com/go-acme/lego/blob/master/providers/dns/cloudflare/cloudflare.toml)
- [x] generate CLI help, documentation, and readme.
- [x] be able to do: _(and put the output of this command to a comment)_
```bash
rm -rf .lego

./lego -m your_email@example.com --dns ionos -d *.example.com -d example.com -s https://acme-staging-v02.api.letsencrypt.org/directory run
```
Note the wildcard domain is important.

@AlexH269 could you test by running the previous command (you have to use a real domain and email)?

Closes #727

### CODE REVIEW

**Comment by jankatins** [comment]
Worked for me :-)

[21:57:58] [1] ✗   IONOS_API_KEY=prefix.secret_long_key dist/lego -m me@domain.net --dns ionos -d '*.domain.de' -d domain.de -s https://acme-staging-v02.api.letsencrypt.org/directory run
2021/01/13 21:58:14 Please review the TOS at https://letsencrypt.org/documents/LE-SA-v1.2-November-15-2017.pdf
Do you accept the TOS? Y/n
Y
2021/01/13 21:58:19 [INFO] acme: Registering account for me@domain.net
!!!! HEADS UP !!!!

Your account credentials have been saved in your Let's Encrypt
configuration directory at "/home/js/external/lego/.lego/accounts".

You should make a secure backup of this folder now. This
configuration directory will also contain certificates and
private keys obtained from Let's Encrypt so making regular
backups of this folder is ideal.
2021/01/13 21:58:20 [INFO] [*.domain.de, domain.de] acme: Obtaining bundled SAN certificate
2021/01/13 21:58:21 [INFO] [*.domain.de] AuthURL: https://acme-staging-v02.api.letsencrypt.org/acme/authz-v3/191008559
2021/01/13 21:58:21 [INFO] [domain.de] AuthURL: https://acme-staging-v02.api.letsencrypt.org/acme/authz-v3/191008560
2021/01/13 21:58:21 [INFO] [*.domain.de] acme: use dns-01 solver
2021/01/13 21:58:21 [INFO] [domain.de] acme: Could not find solver for: tls-alpn-01
2021/01/13 21:58:21 [INFO] [domain.de] acme: Could not find solver for: http-01
2021/01/13 21:58:21 [INFO] [domain.de] acme: use dns-01 solver
2021/01/13 21:58:21 [INFO] [*.domain.de] acme: Preparing to solve DNS-01
2021/01/13 21:58:24 [INFO] [domain.de] acme: Preparing to solve DNS-01
2021/01/13 21:58:25 [INFO] [*.domain.de] acme: Trying to solve DNS-01
2021/01/13 21:58:25 [INFO] [*.domain.de] acme: Checking DNS record propagation using [192.168.181.1:53]
2021/01/13 21:58:27 [INFO] Wait for propagation [timeout: 1m0s, interval: 2s]
2021/01/13 21:58:32 [INFO] [*.domain.de] The server validated our request
2021/01/13 21:58:32 [INFO] [domain.de] acme: Trying to solve DNS-01
2021/01/13 21:58:32 [INFO] [domain.de] acme: Checking DNS record propagation using [192.168.181.1:53]
2021/01/13 21:58:34 [INFO] Wait for propagation [timeout: 1m0s, interval: 2s]
2021/01/13 21:58:35 [INFO] [domain.de] The server validated our request
2021/01/13 21:58:35 [INFO] [*.domain.de] acme: Cleaning DNS-01 challenge
2021/01/13 21:58:36 [INFO] [domain.de] acme: Cleaning DNS-01 challenge
2021/01/13 21:58:37 [INFO] [*.domain.de, jankatins.de] acme: Validations succeeded; requesting certificates
2021/01/13 21:58:38 [INFO] [*.domain.de] Server responded with a certificate.

File: docs/content/dns/zz_gen_ionos.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+```
+
+
+
+
+## Credentials
+
+| Environment Variable Name | Description |
+|-----------------------|-------------|
+| `IONOS_API_KEY` | API Key |
```
Comment (jankatins) [inline docs/content/dns/zz_gen_ionos.md:35]: `API KEY (<prefix>.<secret>)` and maybe a link to https://developer.hosting.ionos.de/docs/getstarted ?

**Comment by ldez** [comment]
@jankatins Thank you :+1:

File: providers/dns/ionos/internal/types.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+	if len(f.errors) == 0 {
+		return nil
+	}
+
+	return &f.errors[0]
+}
+
+// Error defines model for error.
+type Error struct {
+
```
Comment (dmke) [inline providers/dns/ionos/internal/types.go:43]: Here, in `Zone`, and in `CustomerZone` are some extraneous blank lines.

**Review by dmke — APPROVED** [review]
LGTM, but I wouldn't mind to have those pesky blank lines eliminate :-)

**Review by dmke — APPROVED** [review]

**Pull request merged**